### DISCUSSION TITLE
Poor image quality after upgrading to 1.3.0 from older version

### DISCUSSION THREAD

**Discussion by petergaal** [rustdesk/rustdesk#9211] [General]
### Bug Description

Hi. I just recently upgraded to the latest version 1.3.0 from an older version 1.2.3 and I noticed a visible degradation of image quality. The older version stopped to connect for some reason so I installed the latest version, but the image quality is so bad that is almost unreadable. It looks like a heavily compressed image with a lossy compression (like jpeg).

### How to Reproduce

In my case I used the older 1.2.3 version, upgraded to 1.3.0 and the image quality is much worse. Selecting better image quality didn't help.

### Expected Behavior

Better image quality, what is readable

### Operating system(s) on local side and remote side

Windows 10 on both sides

### RustDesk Version(s) on local side and remote side

1.3.0 -> 1.3.0

### Screenshots

This is roughly how it looks like. Areas are blacked out to remove some information, but generally, this is how it can look like. Although it's usually a bit better than this, it's still quite unreadable. This was taken with a phone, as printscreen button is creating a correct image from the remote computer.
![rustdesk-1 3 0](https://github.com/user-attachments/assets/ff8703ee-5a21-4b79-b177-d702218274ab)

**Comment by rustdesk** [comment]
1, How about the networking?

2, Could you show the log following this https://github.com/rustdesk/rustdesk/discussions/8651#discussioncomment-10475692? more logs are also welcome.

**Reply by petergaal** [reply]
Hi. There is no change in network. The locations are quite far away, in different countries in Europe. Both sides are on fiber optics (fiber to the home directly) and everything is on ethernet cable (not wifi). On one side the speed is 500/100 Mbit/s, other side is 80/20 Mbit/s (download/upload). Nothing else was changed, except the software. Also, on other remote software I don't see any problem or image degradation (Anydesk).
RAM usage is normal, there are 4 processes and 3 of them are under 60MB, last one is less than 3MB RAM.
Zipped log folder is attached, I removed one folder from there.
[log.zip](https://github.com/user-attachments/files/16815426/log.zip)

**Reply by rustdesk** [reply]
You'd better also test the same controlled machine with near machine if possible.

**Reply by rustdesk** [reply]
Also please show this.

![image](https://github.com/user-attachments/assets/72ad25a7-3c8b-4fd9-a30d-e6c8dec1a996)

You can see above by.

![image](https://github.com/user-attachments/assets/4c17434f-b14f-4a60-8e01-379f8e739c5f)

**Reply by rustdesk** [reply]
@21pages check the log, it seems related to our QOS.

**Reply by petergaal** [reply]
Here is the picture. You can see the degradation little bit as well. The kB/s are sometimes up, this was when I didn't move a cursor.
Also one thing, when I move the cursor and something get refreshed on the screen (smaller areas), the picture gets better over time. But when there is no change, the same bad quality stays on the picture, until some smaller changes will happen on the screen.
![IMG_0026](https://github.com/user-attachments/assets/2d2330d3-3957-4ddc-946c-bc59d16b716b)
Edit: the picture is taken with phone, as I am not able to get a screenshot with printscreen button with the application open.

**Reply by petergaal** [reply]
When there is some activity (for example now I have something in putty terminal screen, which is black/white mostly), the delay goes to 46 ms. Speed, usually less than 100kB/s, but have seen 124kB/s on more changes, FPS maybe 6 (not many changes are there).

**Reply by 21pages** [reply]
> the picture gets better over time

This is normal.

Qos is not changed, always balanced. You can upload rustdesk recording to tell the quality.

![image](https://github.com/user-attachments/assets/c364579d-9823-482d-a880-e356eacef686)
![image](https://github.com/user-attachments/assets/b5e0e79d-d909-4611-9c73-185b3959e385)

**Reply by rustdesk** [reply]
Interesting, why is this also not clear? Could you send screenshot?

![image](https://github.com/user-attachments/assets/e351bdbe-9f4d-4765-a5b6-c91ea0eb7cb2)

**Reply by petergaal** [reply]
Here I tried to browse random news. I think the quality here is not that bad, sometimes is worse. But you can see how it works.
https://github.com/user-attachments/assets/9b7b8a75-6546-4df3-8630-a4d2e2a87c4d
I don't see the quality screen on the video. How can I include that on any screenshot? The quality screen is good on my side, the low quality is because I took it with the phone.

**Reply by petergaal** [reply]
Here is the screenshot, took with printscreen and cropped just that part.
![rustdesk-screenshot1](https://github.com/user-attachments/assets/73a49176-2ffb-4347-a715-6f9d5358ed9d)

**Reply by 21pages** [reply]
In fact, this is a known issue, just with varying degrees on different computers, different fps and different screen content . 

To prevent blur in dynamic scenes, different bitrate controls may be needed. If additional frames are sent after a dynamic scene, it can quickly recover from blur,  but currently, if the screen is static, the fps is 0, this is a trade-off.

Try selecting `Best` quality or setting custom bitrate, setting custom bitrate may require reconnecting to take effect.

**Reply by petergaal** [reply]
Ok, custom quality improved it, it's not perfect, but readable now. Was this changes recently? Because in the older version, I didn't see this problem.

**Reply by rustdesk** [reply]
> a known issue

We need to solve, because this scenario is common, can not be ignored.

**Reply by melroy89** [reply]
I believe its still not really resolved.

**Reply by thescrapyarduk** [reply]
No, its still laggy. Like the screen display is catching up. I have a 2GB fibre network and cat 7 cables so shouldn't be any slow down with display updates. I previously used AnyDesk that was much faster display updates but laggy network and no sound transmitted

RustDesk, for me, is the better option as the network is faster but display laggy. I can manage with the laggy display as I don't use it for game related more remote computer access on the bartop arcade cabinet and bartop pinball I built. I can do updates without being in front of them

Display is still laggy. Maybe I'll try different compression and see what works best for me. Be nice is there was a RAW option to send display as-is if you have a fast LAN or network
